src/main/java/org/nuxeo/ecm/core/storage/marklogic/MarkLogicRepository.java

```diff
@@ -94,9 +94,9 @@
 
     private static final Function<String, String> ID_FORMATTER = id -> String.format("/%s.xml", id);
 
-    public static final String DB_DEFAULT = "nuxeo";
+    private static final CursorService<ResultSequence, ResultItem> CURSOR_SERVICE = new CursorService<>();
```

Comment:
I don't like that this singleton holding state is held in a static field. For the backport that's acceptable but this should be refactored in master to be a proper Nuxeo service.